src/serlio/materials/MaterialCommand.cpp

```diff
@@ -0,0 +1,178 @@
+/**
+ * Serlio - Esri CityEngine Plugin for Autodesk Maya
+ *
+ * See https://github.com/esri/serlio for build and usage instructions.
+ *
+ * Copyright (c) 2012-2019 Esri R&D Center Zurich
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "materials/StingrayMaterialNode.h"
+#include "materials/ArnoldMaterialNode.h"
+#include "materials/MaterialCommand.h"
+
+#include "utils/MayaUtilities.h"
+
+#include "maya/MArgList.h"
+#include "maya/MFloatPointArray.h"
+#include "maya/MFnMesh.h"
+#include "maya/MGlobal.h"
+#include "maya/MItDependencyGraph.h"
+#include "maya/MItSelectionList.h"
+
+bool MaterialCommand::isUndoable() const {
+	return true;
+}
+
+MStatus MaterialCommand::doIt(const MArgList& argList) {
+	MStatus status;
+
+	if (argList.length() == 0) {
+		cerr << "Material type expected (stingray/arnold)" << endl;
```

Comment:
We should avoid using `cout`/`cerr` in production code as this will go unseen by the user (especially on windows). Also, here it feels redundant as we already have `displayError`. Consider removing the `cerr` lines.

Reply:
removed all occurences